management/src/test/java/org/ehcache/management/providers/statistics/MissRateTest.java

```diff
@@ -0,0 +1,147 @@
+    { newResourcePoolsBuilder().disk(1, MB), Arrays.asList("Disk:MissRate"), Arrays.asList(CACHE_MISS_RATE), CACHE_MISS_RATE },
+
+    //2 tier
+    { newResourcePoolsBuilder().heap(1, MB).offheap(2, MB), Arrays.asList("OnHeap:MissRate","OffHeap:MissRate"), Arrays.asList(3d/seconds,3d/seconds), CACHE_MISS_RATE},
+    { newResourcePoolsBuilder().heap(1, MB).disk(2, MB), Arrays.asList("OnHeap:MissRate","Disk:MissRate"), Arrays.asList(3d/seconds,3d/seconds), CACHE_MISS_RATE},
+    //offheap and disk configuration is not valid.  Throws IllegalStateException no Store.Provider found to handle configured resource types [offheap,disk]
+
+    //3 tier
+    { newResourcePoolsBuilder().heap(1, MB).offheap(2, MB).disk(3, MB), Arrays.asList("OnHeap:MissRate","OffHeap:MissRate","Disk:MissRate"), Arrays.asList(3d/seconds,3d/seconds,3d/seconds), CACHE_MISS_RATE},
+    });
+  }
+
+  public MissRateTest(Builder<? extends ResourcePools> resources, List<String> statNames, List<Double> tierExpectedValues, Double cacheExpectedValue) {
+    this.resources = resources.build();
+    this.statNames = statNames;
+    this.tierExpectedValues = tierExpectedValues;
+    this.cacheExpectedValue = cacheExpectedValue;
+  }
+
+  @Test
+  public void test() throws InterruptedException, IOException {
+
+    CacheManager cacheManager = null;
+
+    try {
+
+      DefaultManagementRegistryConfiguration registryConfiguration = new DefaultManagementRegistryConfiguration().setCacheManagerAlias("myCacheManager");
+      registryConfiguration.addConfiguration(new EhcacheStatisticsProviderConfiguration(1,TimeUnit.MINUTES,100,1,TimeUnit.MILLISECONDS,10,TimeUnit.MINUTES));
+      ManagementRegistryService managementRegistry = new DefaultManagementRegistryService(registryConfiguration);
+
+      CacheConfiguration<Long, String> cacheConfiguration = CacheConfigurationBuilder.newCacheConfigurationBuilder(Long.class, String.class, resources).build();
+
+      cacheManager = CacheManagerBuilder.newCacheManagerBuilder()
+          .withCache("myCache", cacheConfiguration)
+          .using(managementRegistry)
+          .using(new DefaultPersistenceConfiguration(diskPath.newFolder()))
+          .build(true);
+
+      Cache<Long, String> cache = cacheManager.getCache("myCache", Long.class, String.class);
+
+      Context context = StatsUtil.createContext(managementRegistry);
+
+      StatsUtil.triggerStatComputation(managementRegistry, context, "OnHeap:MissRate","OffHeap:MissRate","Disk:MissRate","Cache:MissRate");
+
+      //Put values in cache
+      cache.put(1L, "one");
+      cache.put(2L, "two");
+      cache.put(3L, "three");
+
+      cache.get(4L);//MISS
+      cache.get(5L);//MISS
+      cache.get(6L);//MISS
+
+      //TIER stats
+      for (int i = 0; i < statNames.size(); i++) {
+        double tierMissRate = StatsUtil.getExpectedValueFromRateHistory(statNames.get(i), context, managementRegistry, tierExpectedValues.get(i));
```

Comment:
Please have a look at the clean up I did in #1629, you end up asserting twice the same thing for no added value